src/all_about_leaves.Rmd

```diff
@@ -90,39 +90,98 @@ important_leaf_traits <- leaf_traits %>%
   select(scientificname, name, supermean) %>%
   unique
 
-#The below code accomplishes what the for loop is intended to accomplish.
-
-install.packages('reshape2')
-library(reshape2)
 organized_important_leaf_traits <- dcast(important_leaf_traits, scientificname ~ name, value.var="supermean")
 
- 
+predict_species <- function(c2n_leaf = NA, leaf_biomass_area = NA, leafN = NA, leafP = NA, leaf_respiration_rate_m2 = NA, leaf_longevity = NA, leafC = NA, leaf_biomass_plant = NA, leaf_turnover_rate = NA) {
+  prediction_key = c(c2n_leaf, leaf_biomass_area, leaf_biomass_plant, leaf_longevity, leaf_respiration_rate_m2, leaf_turnover_rate, leafC, leafN, leafP)
+  n = nrow(organized_important_leaf_traits)
```

Comment:
Although I didn't do it in the MH example it is best to explicitly pass any objects you need in the function call. For functions all run in a single script it isn't essential, but in practice would make code easier to understand and more portable.